File: 1027-longest-arithmetic-subsequence/1027-longest-arithmetic-subsequence.py
class Solution:
    def longestArithSeqLength(self, nums: List[int]) -> int:
        dp = {}
        for i in range(1, len(nums)):
            for j in range(0, i):
                diff = nums[i] - nums[j]
                dp[i, diff] = dp.get((j, diff), 1) + 1
        return max(dp.values())

File: 1027-longest-arithmetic-subsequence/1027-longest-arithmetic-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestArithSeqLength = function(nums) {
  const n = nums.length;
  const dp = Array.from({ length: n }, () => new Map());
  let ans = 0;
  
  for (let i = 1; i < n; i++) {
    for (let j = 0; j < i; j++) {
      const diff = nums[i] - nums[j];
      dp[i].set(diff, (dp[j].get(diff) || 1) + 1);
      ans = Math.max(ans, dp[i].get(diff));
    }
  }
  
  return ans;
};